### PR TITLE
Add domain to shred site data warning

### PR DESCRIPTION
**Pull request by cuba**
Resolves https://github.com/brave/brave-browser/issues/39113

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Screenshots
Simulator Screenshot - iPhone 14 Pro - 2024-07-09 at 09 09 06

## Test Plan:
Steps in issue

### CODE REVIEW

File: ios/brave-ios/Sources/BraveShields/ShieldStrings.swift [hunk 1]
```diff
@@ -345,15 +345,28 @@ extension Strings.Shields {
...
+      The placeholder will be replaced with a website domain.
+      """
   )
 
+  /// A message for a confirmation window that appears when a user clicks on 'Shred Data'.
+  /// The provided URL will be used to extract the domain or host.
+  public static func shredThisSitesDataConfirmationMessage(for url: URL) -> String {
+    return String(
+      format: Strings.Shields.shredThisSitesDataConfirmationMessage,
+      url.baseDomain ?? url.host ?? url.absoluteString
```
Comment (iccub) [inline ios/brave-ios/Sources/BraveShields/ShieldStrings.swift:366]: IMO this does not belong here and should be put to View classes or an extension

**Review by iccub — CHANGES_REQUESTED** [review]

File: ios/brave-ios/Sources/BraveShields/ShieldStrings.swift (hunk 1 above)
Comment (cuba) [reply ios/brave-ios/Sources/BraveShields/ShieldStrings.swift:366]: What about `Strings+ShieldsPanelView`?

Comment (iccub) [reply ios/brave-ios/Sources/BraveShields/ShieldStrings.swift:366]: to me this logic can be duplicated across two files as I see it's used in two places. It is a very basic logic to use

File: ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+TabManagerDelegate.swift [hunk 2]
```diff
@@ -438,9 +438,9 @@ extension BrowserViewController: TabManagerDelegate {
         image: UIImage(braveSystemNamed: "leo.shred.data"),
         attributes: .destructive,
         handler: UIAction.deferredActionHandler { [weak self] _ in
-          let alert = UIAlertController.shredDataAlert { _ in
-            guard let tab = self?.tabManager.selectedTab else { return }
-            guard let url = tab.url else { return }
+          guard let tab = self?.tabManager.selectedTab else { return }
+          guard let url = tab.url else { return }
```
Comment (Brandon-T) [inline ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+TabManagerDelegate.swift:442]: Merge the guard statements into one.

File: ios/brave-ios/Sources/Brave/Frontend/Browser/Tabs/TabTray/TabTrayController.swift [hunk 3]
```diff
@@ -704,9 +704,10 @@ class TabTrayController: AuthenticationController {
   }
 
   @objc func shredButtonPressed() {
-    let alert = UIAlertController.shredDataAlert { _ in
-      guard let tab = self.tabManager.selectedTab else { return }
-      guard let url = tab.url else { return }
+    guard let tab = self.tabManager.selectedTab else { return }
+    guard let url = tab.url else { return }
```
Comment (Brandon-T) [inline ios/brave-ios/Sources/Brave/Frontend/Browser/Tabs/TabTray/TabTrayController.swift:708]: Only need one guard statement instead of 2.

**Review by Brandon-T — CHANGES_REQUESTED** [review]

**Review by iccub — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**